src/test/java/de/retest/recheck/review/ignore/PixelDiffFilterTest.java

```diff
@@ -14,7 +14,7 @@
 class PixelDiffFilterTest {
 
 	double pixelDiff = 5.0;
-	Filter cut = new PixelDiffFilter( pixelDiff );
+	Filter cut = new PixelDiffFilter( false, pixelDiff );
```

Comment:
Just for safety, do we want a `new PixelDiffFilter( true, pixelDiff )` for specific checks? Or do you think, we should expand the `PixelDiffFilterLoaderTest` to have it call `save` to ensure formatting?

Also, when do you think, we should use a `@BeforeEach`-method?

Reply:
> Just for safety, do we want a `new PixelDiffFilter( true, pixelDiff )` for specific checks? Or do you think, we should expand the `PixelDiffFilterLoaderTest` to have it call `save` to ensure formatting?

It is currently tested via the `LoadAndSaveFilterWorkerTest`, but I can introduce a separate unit test for the `toString()` method if you want?

> Also, when do you think, we should use a `@BeforeEach`-method?

It boils down to:

```java
class MyTest {
	Object o = new Object();
}
```

vs.:

```java
class MyTest {
	Object o;

	@BeforeEach
	void setUp() {
		o = new Object();
	}
}
```

What is the benefit of the latter if there is no complex setup (e.g. with mocking)? Although one can change the [lifecycle of a JUnit 5 test class](https://blog.codefx.org/libraries/junit-5-basics/#Test-Class-Lifecycle), this probably never happens globally but rather on demand per class.

Our [wiki article](https://retest.atlassian.net/wiki/spaces/RET/pages/160464907/Testing) also suggests no general rule here at the moment, so maybe we should gather some other opinions as well as best practices and discuss this in the retro?

Reply:
> It is currently tested via the `LoadAndSaveFilterWorkerTest`, but I can introduce a separate unit test for the `toString()` method if you want?

Sure, it is up to you, if you want to test the `PixelDiffFilter#toString` or the respective `PixelDiffFilterLoaderTest`. Mainly that we have a more direct test with an appropriate description for this fix.

IMHO, `LoadShouldIgnoreWorkerTest` should be an integration test (but that's separate, too).

Reply:
OK, will add a unit test, we can still keep the integration test. 👍